hubble/executor/helper.py

```diff
@@ -270,6 +270,39 @@ def parse_hub_uri(uri_path: str) -> Tuple[str, str, str, str]:
     return scheme, name, tag, secret
 
 
+def parse_hub_uri(uri_path: str) -> Tuple[str, str, str, str]:
+    """Parse the uri of the Jina Hub executor.
+
+    :param uri_path: the uri of Jina Hub executor
+    :return: a tuple of schema, id, tag, and secret
+    """
+    if uri_path.startswith('jinahub'):
+        return parse_legacy_hub_uri(uri_path)
+
+    parser = urlparse(uri_path)
+    scheme = parser.scheme
+    if scheme not in {'jinaai', 'jinaai+docker', 'jinaai+sandbox'}:
+        raise ValueError(f'{uri_path} is not a valid Hub URI.')
+
+    if not parser.path:
+        raise ValueError(f'{uri_path} is not a valid Hub URI.')
+
+    items = parser.path.strip('/').split(':')
+
+    if len(items) > 2:
+        raise ValueError(f'{uri_path} is not a valid Hub URI.')
+
+    name = items[0].strip('/')
+    tag = items[1].strip('/') if len(items) > 1 else None
+
```

Comment:
can we make a private function out of this few lines ? And add unit test so see if it cover all of the corner case

Reply:
Using a class is even better to encapsulate the different operations on the URI.

Reply:
IMO, it isn't worth the effort. More or less all edge cases have been tested.